cmd/root.go

```diff
@@ -376,20 +376,23 @@ func ExecuteContext(ctx context.Context, rootCmd *cobra.Command, clients *shared
 
 	// The cleanup() method in the root command will invoke via `defer` from within Execute.
 	if err := rootCmd.ExecuteContext(ctx); err != nil {
-		errMsg := err.Error()
-		clients.EventTracker.SetErrorMessage(errMsg)
-		if slackErr, ok := err.(*slackerror.Error); ok {
-			clients.EventTracker.SetErrorCode(slackErr.Code)
-		}
 		if slackerror.Is(err, slackerror.ErrProcessInterrupted) {
 			clients.IO.SetExitCode(iostreams.ExitCancel)
-			clients.IO.PrintDebug(ctx, errMsg)
+			clients.IO.PrintDebug(ctx, err.Error())
 		} else {
+			if slackerror.Is(err, slackerror.ErrSDKHookNotFound) && clients.SDKConfig.Runtime == "" {
+				err = slackerror.New(slackerror.ErrRuntimeNotFound).
+					WithRootCause(slackerror.ToSlackError(err).WithRemediation(""))
+			}
```

Comment:
🔏 note: The `runtime` is unknown to actual hook scripts, so the `runtime_not_found` error can't be determined during hook execution itself...

🤔 question: Does this error handling logic make sense to include at the root command cleanup, or should hook executions be checked for this?